arch/arm/include/armv6-m/irq.h

```diff
@@ -219,8 +219,7 @@ struct xcptcontext
 
 /* Get/set the PRIMASK register */
 
-static inline uint8_t getprimask(void) inline_function;
-static inline uint8_t getprimask(void)
+inline_function static uint8_t getprimask(void)
```

Comment:
I noticed that some functions like this one had "inline" before the name of the function and "inline_function" at the end of the function. It is really strange. Maybe we can try to pass "-std=c89" for some arch that is already following C89 to try to catch these issues. Other option is to find "an old era" C89 compiler for some retro arch and test it.

Reply:
I think it would be a good thing to pass -std=c89 on common files in the precheck. Then if someone inadvertently uses C99 in common files, it will be noticed immediately.

Reply:
(Maybe not just in the precheck, but as a standard part of the build system?)

Reply:
Yes, I think passing -std=c89 for common files is a good idea.

Reply:
> I think it would be a good thing to pass -std=c89 on common files in the precheck. Then if someone inadvertently uses C99 in common files, it will be noticed immediately.

@hartmannathan I have just tested -std=c89 option to check if gcc detects errors in case of mixed declarations and code.
As it turned out, -std=c89 is not enough. Only "-std=c89 -Wpedantic" detects this case:
```
$ gcc -x c -std=c89 -Wpedantic - <<EOF
> int main() {
>   int a;
>   a = 2;
>   int b;
>   return 0;
> }
> EOF
<stdin>: In function ‘main’:
<stdin>:4:3: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
```

Reply:
@a-lunev thanks for letting me know... do you agree that we should use this in the precheck builds (or maybe even in all builds, to catch standard violations as soon as they occur)?

Reply:
> @a-lunev thanks for letting me know... do you agree that we should use this in the precheck builds (or maybe even in all builds, to catch standard violations as soon as they occur)?

Yes, I like this idea for automatic validation by passing -std=c89 option. Though, I do not know what the stage is better (pre-check or a standard part of the build system). I am not yet familiar to all internals how the NuttX build system works. Is the pre-check invoked every time I run "make" command to build NuttX?